doc/nrf/gs_programming.rst

```diff
@@ -119,3 +119,64 @@ After completing the :ref:`manual <build_environment_cli>` or :ref:`automatic <g
+| :file:`zephyr.hex`              | Default full image.                                                                 | Programming non-secure domain (``_ns``) or single-core build targets. |
++---------------------------------+-------------------------------------------------------------------------------------+-----------------------------------------------------------------------+
+| :file:`merged_domain.hex`       | Default full image for multiple cores.                                              | Programming secure domain (``_s``) and multi-core build targets.      |
++---------------------------------+-------------------------------------------------------------------------------------+-----------------------------------------------------------------------+
+| :file:`merged.hex`              | A metadata file used to create the hexidecimal file.                                | Programming multi-core application.                                   |
+|                                 | Not used by Nordic Semiconductor's build targets.                                   |                                                                       |
++---------------------------------+-------------------------------------------------------------------------------------+-----------------------------------------------------------------------+
+| :file:`dfu_application.zip`     | MCUboot compatible update image for both cores.                                     | Using nRF Programmer for Android and iOS, or nRF Connect Programmer.  |
++---------------------------------+-------------------------------------------------------------------------------------+-----------------------------------------------------------------------+
+
+.. output_build_files_table_end
+
+.. _gs_programming_output_artifacts:
+
+Build artifacts
+===============
+
+The following table lists build file artifacts that can be generated when building firmware, but are only used to create the final output build files listed in the table above.
+
++---------------------------------+----------------------------------------------------------------------------------------------------------+
+| File                            | File format                                                                                              |
++=================================+==========================================================================================================+
+| :file:`zephyr.elf`              | A metadata file used to create the :file:`zephyr.hex` file.                                              |
++---------------------------------+----------------------------------------------------------------------------------------------------------+
+| :file:`app_update.bin`          | Application core update file used to create :file:`dfu_application.zip`.                                 |
```

Comment:
We should also state that this is the file that would be transferred in an actual update procedure, as opposed to the hex files we use when we use nrfjprog to emulate an update procedure.